resource-managers/kubernetes/core/pom.xml

```diff
@@ -29,7 +29,7 @@
   <name>Spark Project Kubernetes</name>
   <properties>
     <sbt.project.name>kubernetes</sbt.project.name>
-    <kubernetes.client.version>4.4.2</kubernetes.client.version>
+    <kubernetes.client.version>4.6.0</kubernetes.client.version>
```

Comment:
You need to change the `kubernetes.client.version` in `resource-managers/kubernetes/integration-tests/pom.xml` as well. 

and bump: 
```
    <dependency>
      <groupId>com.squareup.okhttp3</groupId>
      <artifactId>okhttp</artifactId>
      - <version>3.8.1</version>
      + <version>3.12.0</version>
    </dependency>
```
as I believe there is a dependency here https://mvnrepository.com/artifact/io.fabric8/kubernetes-client/4.6.0

Reply:
Ok. Is there a reason why okhttp is added as an explicit dependency(kubernetes-client already brings it)? If we really need a specific version, shouldn't we at least exclude it from `kubernetes-client`? I'm saying this because it's very easy to end up with a unexpected version on the classpath. `kubernetes-client` uses okhttp's version 3.12 since version 4.1.2 but we're only updating it now.

Reply:
That is a good point, @mccheah do we have an opinion on this? I am okay with removing the explicit dependency since `kubernetes-client` already brings it in. If there was a reason, I can't seem to remember, to explicitly state, let's exclude it?

Reply:
This one looks okay to me.

Reply:
agreed we should remove explicit dep to avoid a mismatch failure in the future